test/TestParser.java

```diff
@@ -0,0 +1,134 @@
+import org.junit.rules.*;
+import org.junit.Rule;
+import org.junit.Ignore;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class TestParser {
+   private final String fileName = "test.txt";
+   private final String unicodeString = "Юникодтекст";
+   private final String nonUnicodeString = "Simple text";
+   @Rule
+   public TemporaryFolder folder = new TemporaryFolder();
+   public File saveStringToFile(String str) throws IOException {
+      File f = this.folder.newFile(this.fileName);
+      try {
+         PrintWriter out = new PrintWriter(f.getAbsoluteFile());
+         try {
+            out.print(str);
+         } finally {
+            out.close();
+         }
+      } catch(IOException e) {
+         throw new RuntimeException(e);
+      }
+      return f;
+   }
+   public String readFromFile(File f) throws IOException {
+      String str = "";
+      try {
+         byte[] fArray = Files.readAllBytes(Paths.get(f.getPath()));
+         str = new String(fArray, "UTF-8");
+      } catch (IOException e) {
+         System.err.println(e);
```

Comment:
What is the reason we catch/log? Should not it be logged if test fails of this particular exception anyway?